tcs/csp_solver_core.cpp

```diff
@@ -3976,73 +3354,44 @@ void C_csp_solver::Ssimulate(C_csp_solver::S_sim_setup & sim_setup)
-				T_cold_solved = tol_solved = std::numeric_limits<double>::quiet_NaN();
-				int iter_solved = -1;
-
-				int solver_code = 0;
-				try
-				{
-					solver_code = c_solver.solve(T_cold_guess_low, T_cold_guess_high, 0.0, T_cold_solved, tol_solved, iter_solved);
-				}
-				catch (C_csp_exception)
-				{
-					throw(C_csp_exception(util::format("At time = %lg, C_csp_solver::CR_ON__PC_RM_HI__TES_FULL failed", mc_kernel.mc_sim_info.ms_ts.m_time), ""));
-				}
+                double q_dot_pc_fixed = std::numeric_limits<double>::quiet_NaN();        //[MWt]
+                op_mode_str = "CR_ON__PC_RM_HI__TES_FULL__AUX_OFF";
 
-				if (solver_code != C_monotonic_eq_solver::CONVERGED)
-				{
-					if (solver_code > C_monotonic_eq_solver::CONVERGED && fabs(tol_solved) < 0.1)
-					{
-						std::string msg = util::format("At time = %lg C_csp_solver::CR_ON__PC_RM_HI__TES_FULL failed "
-							"iteration to find the cold HTF temperature to balance energy between the TES and PC only reached a convergence "
-							"= %lg. Check that results at this timestep are not unreasonably biasing total simulation results",
-							mc_kernel.mc_sim_info.ms_ts.m_time / 3600.0, tol_solved);
-						mc_csp_messages.add_message(C_csp_messages::NOTICE, msg);
-					}
-					else
-					{
-						m_is_CR_ON__PC_RM_HI__TES_FULL__AUX_OFF_avail = false;
-						are_models_converged = false;
-						break;
-					}
-				}
+                double defocus_solved = std::numeric_limits<double>::quiet_NaN();
 
-				// Calculate and report mass flow rate balance
-				double m_dot_rec = mc_cr_out_solver.m_m_dot_salt_tot;	//[kg/hr]
-				double m_dot_pc = mc_pc_out_solver.m_m_dot_htf;			//[kg/hr]
-				double m_dot_tes = mc_tes_ch_htf_state.m_m_dot;			//[kg/hr]
+                int mode_code = solve_operating_mode(cr_mode, pc_mode, solver_mode, step_target_mode,
+                    q_dot_pc_fixed, is_defocus, op_mode_str, defocus_solved);
 
-				double m_dot_bal = (m_dot_rec - (m_dot_pc + m_dot_tes)) / m_dot_rec;		//[-]
+                if (mode_code != 0)
+                {
+                    m_is_CR_ON__PC_RM_HI__TES_FULL__AUX_OFF_avail = false;
+                    are_models_converged = false;
+                    break;
+                }
 
-				if (m_dot_bal > 0.0 || (mc_pc_out_solver.m_q_dot_htf - q_pc_max) / q_pc_max > 1.E-3)
+				if ((mc_pc_out_solver.m_q_dot_htf - m_q_dot_pc_max) / m_q_dot_pc_max > 1.E-3)
 				{
 					m_is_CR_ON__PC_RM_HI__TES_FULL__AUX_OFF_avail = false;
 					are_models_converged = false;
 					break;
 				}
 				else if( mc_pc_out_solver.m_q_dot_htf < q_pc_target )
 				{
-					error_msg = util::format("At time = %lg CR_ON__PC_RM_HI__TES_FULL__AUX_OFF method converged to a power cycle"
+					error_msg = util::format("At time = %lg %s method converged to a power cycle"
 						" thermal input less than the target.",
-						mc_kernel.mc_sim_info.ms_ts.m_time / 3600.0);
+						mc_kernel.mc_sim_info.ms_ts.m_time / 3600.0, op_mode_str);
```

Comment:
This is the cause of your Travis CI failing:

error: cannot pass object of non-trivial type 'std::string' (aka 'basic_string<char, char_traits<char>, allocator<char> >') through variadic function; call will abort at runtime [-Wnon-pod-varargs]

                                                mc_kernel.mc_sim_info.ms_ts.m_time / 3600.0, op_mode_str);

Reply:
fixed https://github.com/NREL/ssc/commit/5ac1512d24fb5be4e53657eff494148d3f088df6